aim/sdk/objects/plugins/deeplake_dataset.py

```diff
@@ -0,0 +1,113 @@
+from aim.storage.object import CustomObject
+import deeplake
+import warnings
+import logging
+from deeplake.util.exceptions import ReadOnlyModeError
+
+logger = logging.getLogger(__name__)
+
+
+class UncommittedDatasetWarning(UserWarning):
+    pass
+
+
+@CustomObject.alias('deeplake.dataset')
+class DeeplakeDataset(CustomObject):
+    """
+    Track Activeloop Deeplake Dataset with versioning.
+
+    :param auto_commit: If dataset head node and uncommitted dataset changes are present an auto_commit
+        will trigger a dataset commit `autocommit on aim run` to enable reproducibility of the run,
+        defaults to ``True``.
+
+    :param auto_save_view: Triggers a save of a view if dataset is an unsaved view to enable reproducibility of the run,
+        defaults to ``True``.
+        It is required that the view was not created on a dataset head with changes.
+
+    :raises TypeError: if the dataset is not a deeplake.Dataset
+    :raises ValueError: if the dataset is a view and has uncommitted changes on its head.
+
+    .. code-block:: python
+
+        import deeplake
+
+        from aim.sdk.objects.plugins.deeplake_dataset import DeeplakeDataset
+        from aim.sdk import Run
+
+        # create dataset object
+        ds = deeplake.dataset('hub://activeloop/cifar100-test')
+
+        # log dataset metadata
+        run = Run(system_tracking_interval=None)
+        run['deeplake_ds'] = DeeplakeDataset(ds)
+    """
+    AIM_NAME = 'deeplake.dataset'
+
+    def __init__(self, dataset: deeplake.Dataset, auto_commit: bool = True, auto_save_view: bool = True):
```

Comment:
Thinking loud here; what are the reasonable defaults here? Could the automatic commit/save have some performance implications unexpected for the user?
CC @gorarakelyan

Reply:
That‘s a valid question. The name change happened only roughly half a year ago. But seemingly without downwards compatibility nor official deprecation warning. I should have split the deprecation from the deeplake implementation. Also, since both implementations work in parallel.

Reply:
resolved with https://github.com/aimhubio/aim/pull/2594/commits/99ed7d5fa800bde5a1ea6b664b68bcda6589c293